File: Equipements - Copie (2)/microservices/equip/src/main/java/tn/esprit/equip/Entity/Panne.java
package tn.esprit.equip.Entity;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonBackReference;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.hibernate.annotations.OnDelete;
import org.hibernate.annotations.OnDeleteAction;

import java.util.Date;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Panne {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;


    private String description;
    private Date date;

    @OneToOne
    @JsonIgnoreProperties({"panne", "affectations"})
    Equipement equipement;




    @ManyToOne(fetch = FetchType.EAGER)
    @OnDelete(action = OnDeleteAction.CASCADE)
    @JsonIgnoreProperties({"pannes", "precedent"})
    private EtatEqui etatActuel;

int end=0;


}

File: Frontend/src/app/DSI/equipements/Panne.ts
import { Equip } from "src/app/equipement/equip";
import { EtatEqui } from "./EtatEqui";
import { Dossier } from "src/app/dossier/Dossier";

export class Panne {
id!:number;

description:string="";
date!:Date;
equipement!:Equip;
etatActuel!:EtatEqui|null;
dossier!:Dossier|null;
end!:number;
nomExecuteur!:string;




}
